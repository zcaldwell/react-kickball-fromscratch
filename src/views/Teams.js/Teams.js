import React from 'react';
import { useState, useEffect } from 'react';
import { fetchTeams } from '../../services/teams';
import { Link } from 'react-router-dom';
// import Team from '../Team/Team';

export default function TeamList() {
  const [teams, setTeams] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchTeams();
      setTeams(data);
      console.log(data);
    };
    fetchData();
  }, []);

  return (
    <div className="teams">
      {teams.map((team) => (
        <li key={team.id}>
          <Link key={team.id} to={`/teams/${team.id}`}>
            {/* <Team Team={Team} /> */}
          </Link>
        </li>
      ))}
    </div>
  );
}
