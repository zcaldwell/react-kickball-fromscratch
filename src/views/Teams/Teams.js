import React from 'react';
import { useState, useEffect } from 'react';
import { fetchTeams } from '../../services/teams';
import TeamList from '../../components/TeamList';

export default function Teams() {
  const [teams, setTeams] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchTeams();
      setTeams(data);
      console.log(data);
    };
    fetchData();
  }, []);

  return (
    <>
      <TeamList teams={teams} />
    </>
  );
}
