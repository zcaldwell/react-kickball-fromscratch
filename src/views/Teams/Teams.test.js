import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Teams from './Teams';

test('Players should display all players', async () => {
  const { container } = render(
    <MemoryRouter>
      <Teams />
    </MemoryRouter>
  );
  await screen.findByText('Bridge City Sneakers');
  expect(container).toMatchSnapshot();
});
