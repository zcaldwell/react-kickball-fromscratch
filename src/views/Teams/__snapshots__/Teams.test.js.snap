// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players should display all players 1`] = `
<div>
  <div>
    <li>
      <a
        href="/teams/1"
      >
        Bridge City Sneakers
      </a>
    </li>
    <li>
      <a
        href="/teams/3"
      >
        Lakeville Thunderfeet
      </a>
    </li>
    <li>
      <a
        href="/teams/4"
      >
        Mt. Hoodies
      </a>
    </li>
    <li>
      <a
        href="/teams/2"
      >
        Stumptown Lumberjacks
      </a>
    </li>
    <li>
      <a
        href="/teams/5"
      >
        Vancougers
      </a>
    </li>
  </div>
</div>
`;
